app/main.py

```diff
@@ -1,34 +1,61 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        self.decks = []
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.create_deck()
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def create_deck(self) -> None:
+        if self.start[0] == self.end[0] and self.start[1] == self.end[1]:
+            self.decks.append(Deck(self.start[0], self.end[1]))
+
+        elif self.start[0] != self.end[0]:
+            for i in range(self.start[0], self.end[0] + 1):
+                self.decks.append(Deck(i, self.start[0]))
+
+        elif self.start[1] != self.end[1]:
+            for i in range(self.start[1], self.end[1] + 1):
+                self.decks.append(Deck(self.start[0], i))
+
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> None:
+        self.get_deck(row, column).is_alive = False
+        self.change_is_drowned()
+
+    def change_is_drowned(self) -> None:
+        if any(dec.is_alive for dec in self.decks):
+            self.is_drowned = False
+        else:
+            self.is_drowned = True
```

Comment:
could be simplified to one line 
```suggestion
        self.is_drowned = not any(dec.is_alive for dec in self.decks)
```